collector/textfile.go

```diff
@@ -116,48 +241,19 @@ func (c *textFileCollector) parseTextFiles() []*dto.MetricFamily {
 		}
 	}
 
-	// Export the mtimes of the successful files.
-	if len(mtimes) > 0 {
-		mtimeMetricFamily := dto.MetricFamily{
-			Name:   proto.String("node_textfile_mtime"),
-			Help:   proto.String("Unixtime mtime of textfiles successfully read."),
-			Type:   dto.MetricType_GAUGE.Enum(),
-			Metric: []*dto.Metric{},
-		}
+	convertMetricFamilies(metricFamilies, ch)
 
-		// Sorting is needed for predictable output comparison in tests.
-		filenames := make([]string, 0, len(mtimes))
-		for filename := range mtimes {
-			filenames = append(filenames, filename)
-		}
-		sort.Strings(filenames)
+	exportMTimes(c, mtimes, ch)
```

Comment:
If you're passing in the collector itself as the first arg, you'll want to just define it was a method of the collector (so then you get `c.exportMTimes(mtimes, ch)`.